src/goto-symex/symex_complexity.cpp

```diff
@@ -0,0 +1,229 @@
+/*******************************************************************\
+
+Module: Symbolic Execution
+
+Author: John Dumbell
+
+\*******************************************************************/
+
+#include "symex_complexity.h"
+#include "goto_symex_state.h"
+#include <cmath>
+
+symex_complexityt::symex_complexityt(
+  message_handlert &logger,
+  const optionst &options)
+  : log(logger)
+{
+  std::size_t limit = options.get_signed_int_option("symex-complexity-limit");
+  if((complexity_active = limit > 0))
+  {
+    // This gives a curve that allows low limits to be rightly restrictive,
+    // while larger numbers are very large to represent what you're asking.
```

Comment:
what you're asking?

Reply:
Was a rather ambiguous reference to the idea that asking for complexity 6 is not as simple as stating that you're allowing double the complexity of 3. Each number is an individual statement of how much complexity you want in a vague curve shape and not directly related to the last (or future) beyond the idea that 1 = super strict, 20 = super lenient.